tests/components/skipControls-test.js

```diff
@@ -0,0 +1,340 @@
+          duration: 60,
+          scrubberBar: {
+            isHovering: false
+          }
+        },
+        rewindOrRequestPreviousVideo: () => {},
+        requestNextVideo: () => {},
+        setFocusedControl: () => {},
+        cancelTimer: () => {},
+        startHideControlBarTimer: () => {}
+      },
+      a11yControls: {
+        seekBy: () => {}
+      }
+    };
+  });
+
+  it('should render a SkipControls', function() {
+    renderComponent();
+    expect(component).toBeTruthy();
+    expect(wrapper.find('.oo-skip-controls').length).toBe(1);
+  });
+
+  it('should NOT render SkipControls when all buttons are disabled', function() {
+    props.skinConfig.skipControls.enabled = true;
+    props.skinConfig.skipControls.buttons.previousVideo.enabled = false;
+    props.skinConfig.skipControls.buttons.skipBackward.enabled = false;
+    props.skinConfig.skipControls.buttons.skipForward.enabled = false;
+    props.skinConfig.skipControls.buttons.nextVideo.enabled = false;
+    renderComponent();
+    expect(wrapper.find('.oo-skip-controls').length).toBe(0);
+  });
+
+  it('should call onMount callback with clientRect of DOM element', function() {
+    let clientRect;
+    props.onMount = (rect) => {
+      clientRect = rect;
+    };
+    renderComponent();
+    expect(clientRect).toEqual({
+      bottom: 0,
+      height: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+      width: 0
+    });
+  });
+
+  it('should cancel auto-hide timer on mouse enter', function() {
+    const spy = jest.spyOn(props.controller, 'cancelTimer');
+    renderComponent();
+    expect(spy.mock.calls.length).toBe(0);
+    wrapper.simulate('mouseEnter');
+    expect(spy).toHaveBeenCalled();
+    expect(spy.mock.calls.length).toBe(1);
+    spy.mockRestore();
+  });
+
+  it('should correctly determine whether video is at live edge', function() {
+    props.controller.state.isLiveStream = true;
+    props.controller.state.duration = 100;
+    props.currentPlayhead = 100;
+    renderComponent();
+    expect(component.isAtLiveEdge()).toBe(true);
+    props.controller.state.isLiveStream = false;
+    props.controller.state.duration = 100;
+    props.currentPlayhead = 100;
+    renderComponent();
+    expect(component.isAtLiveEdge()).toBe(false);
+    props.controller.state.isLiveStream = true;
+    props.controller.state.duration = 100;
+    props.currentPlayhead = 0;
+    renderComponent();
+    expect(component.isAtLiveEdge()).toBe(false);
+  });
+
+  it('should set inactive class when component is inactive', function() {
+    props.isInactive = true;
+    renderComponent();
+    expect(wrapper.find('.oo-skip-controls.oo-inactive').length).toBe(1);
+  });
+
+  it('should set pointer events to none on buttons when component is inactive', function() {
+    enableAllButtons();
+    props.isInactive = true;
+    renderComponent();
+    wrapper.find(ControlButton).forEach((button) => {
+      expect(button.props().style).toEqual({ pointerEvents: 'none' });
```

Comment:
Should we do a contains check instead of equality in case this style prop definition gets expanded?

Reply:
Updated